### ISSUE TITLE
Gatsby Develop instances with "watch" and "overlay" on revert to out-of-date Sanity content upon publication

### ISSUE THREAD

**Issue by kkgthb** [sanity-io/gatsby-source-sanity#87]
I'm playing with "preview servers" for Gatsby with Sanity as a data source _(e.g. Gatsby in "develop" mode hosted on Heroku)_.

When my application host runs `npm start`, it fires up `gatsby develop` with `SANITY_OVERLAY_DRAFTS` true, `SANITY_WATCH_MODE` true, and `SANITY_READ_TOKEN` populated, as well as with `ENABLE_GATSBY_REFRESH_ENDPOINT` true.  `gatsby-config.js` grabs these environment variables and configures `gatsby-source-sanity`.

My `gatsby-node.js` is as follows:

```javascript
const path = require(`path`)

exports.createPages = async ({ graphql, getNode, actions }) => {
  const { createPage } = actions
  const queryResult = await graphql(`
    query {
      allSanityXyzzy(filter: {slug: {current: {ne: null}}}) {
          edges {
            node {
              template,
              message,
              slug {
                current
              }
            }
          }
      }
    }
  `)
  nodes = queryResult.data.allSanityXyzzy.edges
  nodes.forEach(({ node }) => {
    createPage({
      path: node.slug.current,
      component: path.resolve(`./src/templates/${node.template}.js`),
      context: {
        message: node.message,
      },
    })
  })
};
```

My `/src/templates/xyzzy.js` is as follows:

```javascript
import React from "react"

export default function Xyzzy({ pageContext }) {
  return (
    <div>
	  {pageContext.message}
    </div>
  )
}
```

So, as you can see, I don't have a fresh GraphQL attached to my page template.  Maybe I need to?  I'm new to Gatsby/React and was just playing with getting things working at all, and it didn't seem to be strictly necessary just to get a "hello world" on my homepage.

I've noticed that if I fire up a "develop" server with a single `Xyzzy` piece of content, the corresponding page for that content does indeed update as I type into its "`message`" in Sanity Studio.

However, the moment I "publish" the content in Studio, `https://preview.mysite.com/the_slug` reverts to displaying the value of `message` as seen by Gastby when the server spun up -- even if I have a Sanity webhook POSTing to `https://preview.mysite.com/__refresh`.

That is, unless there's still another draft sitting around somewhere in Sanity.  _(Not sure if it has to be a draft in the same object or if it can be a draft anywhere in Sanity -- right now, I'm testing with a one-object schema.)_

If there is, `/__refresh` webhook-based rebuilds "stick."

Even if I then delete the "spare draft."

It's just when there are _no_ drafts in Sanity that a "publish" `/__refresh`-based rebuild reverts to "old" content.  _(Note that if something else "stuck" sometime while the Gatsby-develop server was running, `https://preview.mysite.com/the_slug` will display a message as seen in the last "stuck" publish, not the even-older publish from server-start time.)_

Note:  without making use of `/__refresh`, publication "reverts" no matter whether there is draft content remaining in Sanity or not.  Not sure if this is a bug or a feature.  :)  _(It'd certainly be lovely if it could do something besides "revert," since leaving the extremely guessable `https://preview.mysite.com/__refresh` open to the world exposes me to DDOS / random-high-bill attacks from strangers.)_

But for sure, even `/__refresh` reverting unless there is some piece of draft content remaining in Sanity seems like it must be a bug.

_(That said, please don't take away my workaround and leave me with **no** way of putting fresh content on a page!)_  ;)

**Comment by oalbacha** [comment]
I think I am having the same problem. I see the changes as I make them in the studio through live preview but they disappear as soon as publish

**Comment by DhillonN** [comment]
same issue here

**Comment by destructorboy** [comment]
+1 Same, I've turned off `overlayDrafts` for now.

**Comment by oalbacha** [comment]
It takes a long time to restart the server and rebuild every time. It’s making the DX pretty painful 😖

**Comment by doigo** [comment]
+1 Same issue is there any work around for this ?

**Comment by kenjonespizza** [comment]
+1 Same issue here.  Any fix in place?

**Comment by aaronadamsCA** [comment]
Adding my +1 here. It's hard to continue adopting a commercial CMS when some of the basics are broken and tickets sit open for half a year without a response ☹ continuing to feel like I picked the wrong CMS provider.

**Comment by c-Nosetab** [comment]
+1 to this issue. Really hurting development

**Comment by simonlagneaux** [comment]
+1 Same issue, very frustrating.

**Comment by mthomas100** [comment]
+1 Same issue

**Comment by bjoerge** [comment]
This should be fixed now in [`v7.0.0`](https://github.com/sanity-io/gatsby-source-sanity/blob/main/CHANGELOG.md#700). Thanks for your patience!
